frankenstein/chunk_store.go

```diff
@@ -247,106 +247,107 @@ func (c *AWSChunkStore) lookupChunks(from, through model.Time, matchers []*metri
 	incomingErrors := make(chan error)
 	buckets := bigBuckets(from, through)
 	for _, hour := range buckets {
-		for _, matcher := range matchers {
-			// TODO: probably better to abort everything using contexts on the first error.
-			go c.lookupChunksFor(hour, metricName, matcher, incomingChunkSets, incomingErrors)
-		}
+		// TODO: probably better to abort everything using contexts on the first error.
+		go c.lookupChunksFor(hour, metricName, matchers, incomingChunkSets, incomingErrors)
 	}
 
-	chunkSets := [][]Chunk{}
+	chunkSets := []Chunk{}
 	errors := []error{}
 	for i := 0; i < len(buckets)*len(matchers); i++ {
 		select {
 		case chunkSet := <-incomingChunkSets:
-			chunkSets = append(chunkSets, chunkSet)
+			chunkSets = append(chunkSets, chunkSet...)
 		case err := <-incomingErrors:
 			errors = append(errors, err)
 		}
 	}
 	if len(errors) > 0 {
 		return nil, errors[0]
 	}
-	return nWayIntersect(chunkSets), nil
+	return chunkSets, nil
 }
 
-func (c *AWSChunkStore) lookupChunksFor(hour int64, metricName model.LabelValue, matcher *metric.LabelMatcher, incomingChunkSets chan []Chunk, incomingErrors chan error) {
-	// TODO build support for other matchers
-	if matcher.Type != metric.Equal {
-		incomingErrors <- fmt.Errorf("%s matcher not supported yet", matcher.Type)
-		return
-	}
+func (c *AWSChunkStore) lookupChunksFor(hour int64, metricName model.LabelValue, matchers []*metric.LabelMatcher, incomingChunkSets chan []Chunk, incomingErrors chan error) {
```

Comment:
As it stands this will lookup each matcher sequentially right?  Will probably need parallelising, but not immediately.